src/test/java/seedu/address/logic/parser/AddOrderCommandParserTest.java

```diff
@@ -0,0 +1,104 @@
+import static seedu.address.logic.commands.CommandTestUtil.VALID_NAME_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_PHONE_AMY;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.testutil.TypicalOrders.AMY;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.AddOrderCommand;
+import seedu.address.model.order.Order;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Phone;
+import seedu.address.testutil.OrderBuilder;
+
+
+public class AddOrderCommandParserTest {
+    private AddOrderCommandParser parser = new AddOrderCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        Order expectedOrder = new OrderBuilder(AMY).build();
+
+        // whitespace only preamble
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + NAME_DESC_AMY + PHONE_DESC_AMY
+                + ADDRESS_DESC_AMY + DETAILS_DESC_AMY, new AddOrderCommand(expectedOrder));
+
+        // multiple names - last name accepted
+        assertParseSuccess(parser, NAME_DESC_BOB + NAME_DESC_AMY + PHONE_DESC_AMY
+                + ADDRESS_DESC_AMY + DETAILS_DESC_AMY, new AddOrderCommand(expectedOrder));
+
+        // multiple phones - last phone accepted
+        assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_BOB + PHONE_DESC_AMY
+                + ADDRESS_DESC_AMY + DETAILS_DESC_AMY, new AddOrderCommand(expectedOrder));
+
+        // multiple addresses - last address accepted
+        assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_AMY + ADDRESS_DESC_BOB
+                + ADDRESS_DESC_AMY + DETAILS_DESC_AMY, new AddOrderCommand(expectedOrder));
+    }
+
+    @Test
+    public void parse_compulsoryFieldMissing_failure() {
+        String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddOrderCommand.MESSAGE_USAGE);
+
+        // missing name prefix
+        assertParseFailure(parser, VALID_NAME_AMY + PHONE_DESC_AMY + ADDRESS_DESC_AMY + DETAILS_DESC_AMY,
+                expectedMessage);
+
+        // missing phone prefix
+        assertParseFailure(parser, NAME_DESC_AMY + VALID_PHONE_AMY + ADDRESS_DESC_AMY + DETAILS_DESC_AMY,
+                expectedMessage);
+
+        // missing address prefix
+        assertParseFailure(parser, NAME_DESC_AMY + PHONE_DESC_AMY + VALID_ADDRESS_AMY + DETAILS_DESC_AMY,
+                expectedMessage);
+
+        // missing details prefix
+        assertParseFailure(parser, NAME_DESC_AMY + PHONE_DESC_AMY + ADDRESS_DESC_AMY + VALID_DETAILS_AMY,
+                expectedMessage);
+
+        // all prefixes missing
+        assertParseFailure(parser, VALID_NAME_AMY + VALID_PHONE_AMY + VALID_ADDRESS_AMY + VALID_DETAILS_AMY,
+                expectedMessage);
+    }
+
+    @Test
+    public void parse_invalidValue_failure() {
```

Comment:
A great test to have! I'll most likely implement this for editordercommandtest later on as well.